bin/seed_experience_index.py

```diff
@@ -0,0 +1,108 @@
+
+from sqlmodel import Session, delete
+
+from warren.db import get_session
+from warren.xi.models import (
+    Experience,
+    Relation,
+)
+
+
+def seed_experience_index():
+    """Seed the Experience Index with mocked data.
+
+    The mocked data have to align with the Learning Record Store (LRS).
+    Note: If there are updates to the LRS fixtures, manually update the IRI accordingly.
+    """
+
+    # Get a database session
+    session = next(get_session())
+
+    # Pass the database session to the factory
+    ExperienceFactory.__session__ = session
+
+    # Restore pristine state of the Experience Index
+    session.exec(delete(Relation))
+    session.exec(delete(Experience))
+    session.commit()
+
+    # Declare mocked data
+    mocked_data = [
+        {
+            "iri": "uuid://0aecfa93-cef3-45ae-b7f5-a603e9e45f50",
+            "title": '{"en": "Lesson 1"}',
+            "structure": "atomic",
+            "aggregation_level": 1,
+        },
+        {
+            "iri": "uuid://1c0c127a-f121-4bd1-8db6-918605c2645d",
+            "title": '{"en":"Lesson 2"}',
+            "structure": "atomic",
+            "aggregation_level": 1,
+        },
+        {
+            "iri": "uuid://541dab6b-50ae-4444-b230-494f0621f132",
+            "title": '{"en":"Lesson 3"}',
+            "structure": "atomic",
+            "aggregation_level": 1,
+        },
+        {
+            "iri": "uuid://69d32ad5-3af5-4160-a995-87e09da6865c",
+            "title": '{"en":"Lesson 4"}',
+            "structure": "atomic",
+            "aggregation_level": 1,
+        },
+        {
+            "iri": "uuid://7d4f3c70-1e79-4243-9b7d-166076ce8bfb",
+            "title": '{"en":"Lesson 5"}',
+            "structure": "atomic",
+            "aggregation_level": 1,
+        },
+        {
+            "iri": "uuid://8d386f48-3baa-4acf-8a46-0f2be4ae243e",
+            "title": '{"en":"Lesson 6"}',
+            "structure": "atomic",
+            "aggregation_level": 1,
+        },
+        {
+            "iri": "uuid://b172ec09-97ec-4651-bc57-6eabebf47ed0",
+            "title": '{"en":"Lesson 7"}',
+            "structure": "atomic",
+            "aggregation_level": 1,
+        },
+        {
+            "iri": "uuid://d613b564-5d18-4238-a69c-0fc8cee5d0e7",
+            "title": '{"en":"Lesson 8"}',
+            "structure": "atomic",
+            "aggregation_level": 1,
+        },
+        {
+            "iri": "uuid://dd38149d-956a-483d-8975-c1506de1e1a9",
+            "title": '{"en":"Lesson 9"}',
+            "structure": "atomic",
+            "aggregation_level": 1,
+        },
+        {
+            "iri": "uuid://e151ee65-7a72-478c-ac57-8a02f19e748b",
+            "title": '{"en":"Lesson 10"}',
+            "structure": "atomic",
+            "aggregation_level": 1,
+        },
+    ]
+
+    # Seed the Experience Index with mocked data
+    session.add_all(
+        ExperienceFactory.build(**r) for r in mocked_data
+    )
```

Comment:
A more concise way would be to create `mocked_data` as a `(iri, title)` tuple list and hard code `structure` and `aggregation_level`:

```suggestion
    session.add_all(
        ExperienceFactory.build(
            iri=iri, 
            title= f'{{"en": "{title}"}}',
            structure="atomic", 
            aggregation_level=1
        ) for iri, title in mocked_data
    )
```

Reply:
Or even all at once:

```python
    session.add_all(
        ExperienceFactory.build(
            iri=iri, 
            title= f'{{"en": "lesson {i}"}}',
            structure="atomic", 
            aggregation_level=1
        ) for i, iri in enumerate(
            [
                "uuid://0aecfa93-cef3-45ae-b7f5-a603e9e45f50",
                "uuid://1c0c127a-f121-4bd1-8db6-918605c2645d",
                "uuid://541dab6b-50ae-4444-b230-494f0621f132",
                "uuid://69d32ad5-3af5-4160-a995-87e09da6865c",
                "uuid://7d4f3c70-1e79-4243-9b7d-166076ce8bfb",
                "uuid://8d386f48-3baa-4acf-8a46-0f2be4ae243e",
                "uuid://b172ec09-97ec-4651-bc57-6eabebf47ed0",
                "uuid://d613b564-5d18-4238-a69c-0fc8cee5d0e7",
                "uuid://dd38149d-956a-483d-8975-c1506de1e1a9",
                "uuid://e151ee65-7a72-478c-ac57-8a02f19e748b",
            ]
        )
    )
```

Reply:
Yeah I thought about it, but at first I wanted to extract any data to a dedicated file. I'll for sure refactor it.